File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int min = prices[0];
        int profit = 0;
        for (int i = 0; i < prices.Length; i++){
            int curr_prof  =  prices[i] - min;
            min = Math.Min(min,prices[i]);
            profit = Math.Max(profit, curr_prof);
        }
        return profit;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0;
        int min = prices[0];
        for (int i = 0; i < prices.length;i++){
            int curr_pro = prices[i] - min;
            profit = Math.max(profit, curr_pro);
            min = Math.min(min, prices[i]);
        }
        return profit;
        
    }
}